chunky/src/java/se/llbit/chunky/world/Dimension.java

```diff
@@ -0,0 +1,312 @@
+package se.llbit.chunky.world;
+
+import it.unimi.dsi.fastutil.longs.Long2ObjectMap;
+import it.unimi.dsi.fastutil.longs.Long2ObjectOpenHashMap;
+import se.llbit.chunky.PersistentSettings;
+import se.llbit.chunky.chunk.ChunkData;
+import se.llbit.chunky.chunk.GenericChunkData;
+import se.llbit.chunky.chunk.SimpleChunkData;
+import se.llbit.chunky.map.MapView;
+import se.llbit.chunky.map.WorldMapLoader;
+import se.llbit.chunky.entity.PlayerEntity;
+import se.llbit.chunky.world.listeners.ChunkDeletionListener;
+import se.llbit.chunky.world.listeners.ChunkTopographyListener;
+import se.llbit.chunky.world.listeners.ChunkUpdateListener;
+import se.llbit.chunky.world.region.*;
+import se.llbit.math.Vector3;
+import se.llbit.math.Vector3i;
+import se.llbit.util.annotation.Nullable;
+
+import java.io.File;
+import java.util.*;
+
+/**
+ *
+ */
+public class Dimension {
+  private final World world;
+
+  protected final Long2ObjectMap<Region> regionMap = new Long2ObjectOpenHashMap<>();
+
+  protected final File dimensionDirectory;
+  private Set<PlayerEntityData> playerEntities;
+
+  private final Heightmap heightmap = new Heightmap();
+
+  private final int dimensionId;
+
+  private final Collection<ChunkDeletionListener> chunkDeletionListeners = new LinkedList<>();
+  private final Collection<ChunkTopographyListener> chunkTopographyListeners = new LinkedList<>();
+  private final Collection<ChunkUpdateListener> chunkUpdateListeners = new LinkedList<>();
+
+  private Vector3i spawnPos = null;
+
+  /** Timestamp for level.dat when player data was last loaded. */
+  private long timestamp;
+
+  /**
+   * @param dimensionDirectory Minecraft world directory.
+   * @param timestamp
+   */
+  protected Dimension(World world, int dimensionId, File dimensionDirectory, Set<PlayerEntityData> playerEntities, long timestamp) {
+    this.world = world;
+    this.dimensionId = dimensionId;
+    this.dimensionDirectory = dimensionDirectory;
+    this.playerEntities = playerEntities;
+    this.timestamp = timestamp;
+  }
+
+  public int dimensionId() {
```

Comment:
```suggestion
  public int getDimensionId() {
```

Reply:
Also, is a `int` really the API we want to have here? Is that compatible with Bedrock?

Reply:
No, `int` is definitely not the API we want, but changing this requires a little more fiddling with internals of other classes, this will be the next PR